kornia/feature/matching.py

```diff
@@ -4,24 +4,29 @@
 
 def match_nn(desc1: torch.Tensor, desc2: torch.Tensor,
              dm: Optional[torch.Tensor] = None) -> Tuple[torch.Tensor, torch.Tensor]:
-    """Function, which finds nearest neightbors in desc2 for each vector in desc1.
+    r"""Function, which finds nearest neightbors in desc2 for each vector in desc1.
+
     If the distance matrix dm is not provided, torch.cdist(desc1, desc2) is used.
+
     Args:
         desc1 (torch.Tensor): Batch of descriptors of a shape :math:`(B1, D)`.
         desc2 (torch.Tensor): Batch of descriptors of a shape :math:`(B2, D)`.
-        dm (torch.Tensor): optional. Tensor containing the distances from each descriptor in desc1 to each descriptor in
-        desc2, shape of :math:`(B1, B2)`.
+        dm (torch.Tensor, optional): Tensor containing the distances from each descriptor in desc1
+          to each descriptor in desc2, shape of :math:`(B1, B2)`.
 
-    Return:
-        torch.Tensor: Descriptor distance of matching descriptors, shape of. :math:`(B1, 1)
-        torch.Tensor: Long tensor indexes of matching descriptors in desc1 and desc2. Shape: :math:`(B1, 2)
+    Returns:
+        Tuple[torch.Tensor, torch.Tensor]:
```

Comment:
@ducha-aiki I found this way to properly return tuple that renders well with sphinx. Could you take a look at the whole features module whenever you have time. The docs seems somehow incomplete or broken.